src/Serilog.Sinks.AzureTableStorage/Sinks/AzureTableStorage/AzureBatchingTableStorageSink.cs

```diff
@@ -86,7 +88,9 @@ protected override void EmitBatch(IEnumerable<LogEvent> events)
 
                 _batchRowId++;
             }
-            _table.ExecuteBatch(operation);
+
+            _table.ExecuteBatchAsync(operation)
```

Comment:
That would be ideal because then these possible deadlocks are avoided altogether (and we remove the infinite timeout :)